tactics/redexpr.ml

```diff
@@ -99,7 +99,12 @@ let disch_ref ref =
   match ref with
   | Evaluable.EvalConstRef c -> Some ref
   | Evaluable.EvalProjectionRef p ->
-    let p = Lib.discharge_proj_repr p in
+    let p =
+      if Lib.is_in_section (Names.GlobRef.IndRef (Names.Projection.Repr.inductive p)) then
+        Lib.discharge_proj_repr p
```

Comment:
possibly the change should be in `Lib.discharge_proj_repr` not in its callers
indeed coercion is one of the other users and has the same bug with
~~~
Set Primitive Projections.

Record test : Type := { f : nat }.
Section foo.
  Coercion f : test >-> nat.
End foo.
~~~

Reply:
Oh, I should have checked. I saw this code and thought that there was one client that didn't need the check so I'd better not change `Lib.discharge_proj_repr`.

Reply:
I changed the API. The original function still exists under a different name and could be exposed if necessary.